logstash-core/src/test/java/org/logstash/ClonerTest.java

```diff
@@ -35,12 +35,28 @@ public void testRubyStringCloning() {
         // Check object identity
         assertTrue(result != original);
 
-        // Check different underlying bytes
-        assertTrue(result.getByteList() != original.getByteList());
-
         // Check string equality
         assertEquals(result, original);
 
         assertEquals(javaString, result.asJavaString());
+
+        // NOTE: this is an implementation detail or the underlying sharing:
+        assertTrue(result.getByteList() == original.getByteList());
```

Comment:
I _really_ don't like testing against implementation details like this. 

We need to validate a set list of things about our cloner:
 - when copied, the original and the clone need to be _equivalent_ (this can include anything from testing their `RubyString#==`, the equivalence of the objects returned by `RubyString#asJavaString`)
 - a copy that is modified does not modify the original
 - an original that has been copied and subsequently modified does not modify the copy

If we _do_ test this implementation detail, I believe it should be a separate test that is clearly labeled as a memory optimization so that if/when implementation details change and the test breaks, we can be confident in our ability to rework it to achieve the same goals.

Reply:
added a separate test